wagtail_ab_testing/static_src/components/GoalSelector/index.tsx

```diff
@@ -0,0 +1,171 @@
+import React, { FunctionComponent, useEffect } from 'react';
+import ReactDOM from 'react-dom';
+import styled from 'styled-components';
+
+// A label element without Wagtail's 'float: left;' rule applied
+const SunkenLabel = styled.label`
+    float: unset;
+    width: unset;
+`;
+
+interface PageAPI {
+    id: number;
+    meta: {
+        type: string;
+    };
+    title: string;
+}
+
+interface GoalPageSelectorProps {
+    testPageId: number;
+    onChangeSelectedPage(pageInfo: PageAPI | null): void;
+}
+
+const GoalPageSelector: FunctionComponent<GoalPageSelectorProps> = ({
+    testPageId,
+    onChangeSelectedPage
+}) => {
+    const [selectedPageId, setSelectedPageId] = React.useState<number | null>(
+        null
+    );
+    const [
+        selectedPageInfo,
+        setSelectedPageInfo
+    ] = React.useState<PageAPI | null>(null);
+
+    // Fetch info about the page whenever the selected page ID is changed
+    useEffect(() => {
+        if (selectedPageId) {
+            fetch(`/admin/api/main/pages/${selectedPageId}/`)
```

Comment:
What if `/admin/` isnt where the admin is setup?

Reply:
Thanks! This is fixed now